common/djangoapps/cors_csrf/middleware.py

```diff
@@ -22,35 +47,76 @@
 
 from django.conf import settings
 from django.middleware.csrf import CsrfViewMiddleware
+from django.core.exceptions import MiddlewareNotUsed, ImproperlyConfigured
 
 log = logging.getLogger(__name__)
 
 
+def is_cross_domain_request_allowed(request):
+    """Check whether we should allow the cross-domain request.
+
+    We allow a cross-domain request only if:
+
+    1) The request has an http referer header.
+    2) The request is made securely and the referer has "https://" as the protocol.
+    3) The referer domain has been whitelisted.
+
+    Arguments:
+        request (HttpRequest)
+
+    Returns:
+        bool
+
+    """
+    referer = request.META.get('HTTP_REFERER')
+    if not referer:
+        log.debug("Request does not have a referer, so it is not a cross-domain request.")
+        return False
```

Comment:
Strict enforcement of the referrer would only work over HTTPS, no? Over HTTP the referrer is often not available (cf the original CORS middleware docstring, which points to this paper as the justification for enforcing the referrer only over HTTPS: https://sparrow.ece.cmu.edu/group/731-s11/readings/csrf.pdf ). So this would work well on edx.org, but likely not that well on open edX sites using HTTP - some users would get 405s which would be confusing to the site admin. 

Since you mandate the use of HTTPS for this feature below, it might be worth moving the check for  `request.is_secure()` before the referer check - that would ensure a more explicit message is being logged in these cases?

It would also make sense to not allow this check to be overridden by `CORS_ALLOW_INSECURE`, since the referrer check is not going to work for all users then, and might only add confusion. If you want to keep this option for development environments, it might be worth making it explicit that it should not be used in production.